functions/cloudflare_b2_proxy.js

```diff
@@ -0,0 +1,113 @@
+ */
+
+export default {
+    async fetch(request, env, ctx) {
+      const corsHeaders = {
+        'Access-Control-Allow-Origin': 'https://app.kendra.io',
+        'Access-Control-Allow-Methods': 'GET, OPTIONS',
+        'Access-Control-Allow-Headers': 'Content-Type',
+      };
+  
+      // Handle OPTIONS request
+      if (request.method === 'OPTIONS') {
+        return new Response(null, { headers: corsHeaders });
+      }
+  
+      const url = new URL(request.url);
+      
+      // Return hello world if no path specified
+      if (url.pathname === "/" || !url.pathname) {
+        return new Response("Hello World!", {
+          headers: { 
+            "content-type": "text/plain",
+            ...corsHeaders
+          },
+        });
+      }
+  
+      const fileName = url.pathname.split("/").pop();
+  
+      // Get configuration from environment variables
+      const B2_APPLICATION_KEY_ID = env.B2_APPLICATION_KEY_ID;
+      const B2_APPLICATION_KEY = env.B2_APPLICATION_KEY;
+      const B2_BUCKET_ID = env.B2_BUCKET_ID;
+      const B2_BUCKET_NAME = env.B2_BUCKET_NAME;
+  
+      console.log(`Processing request for file: ${fileName}`);
+      console.log(`Using bucket: ${B2_BUCKET_NAME} (${B2_BUCKET_ID})`);
+  
+      // Step 1: Authorise with Backblaze B2
+      const authResponse = await fetch('https://api.backblazeb2.com/b2api/v2/b2_authorize_account', {
+        headers: {
+          Authorization: 'Basic ' + btoa(B2_APPLICATION_KEY_ID + ':' + B2_APPLICATION_KEY),
+        }
+      });
+  
+      if (!authResponse.ok) {
+        console.log(`B2 authorisation failed with status: ${authResponse.status}`);
+        return new Response('Error authorising B2 account', { 
+          status: 500,
+          headers: corsHeaders
+        });
+      }
+      console.log('B2 authorisation successful');
+  
+      const authData = await authResponse.json();
+      const apiUrl = authData.apiUrl;
+  
+      // Step 2: Get Download Authorisation
+      const downloadAuthorisation = await fetch(`${apiUrl}/b2api/v2/b2_get_download_authorization`, {
+        method: 'POST',
+        headers: {
+          Authorization: authData.authorizationToken,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          bucketId: B2_BUCKET_ID,
+          fileNamePrefix: fileName,
+          validDurationInSeconds: 360000,  // 100 hour expiry
+        }),
+      });
+  
+      if (!downloadAuthorisation.ok) {
+        console.error(`Download authorisation failed with status: ${downloadAuthorisation.status}`);
+        return new Response('Error generating signed URL', { 
+          status: 500,
+          headers: corsHeaders
+        });
+      }
+      console.log('Download authorisation successful');
+  
+      const signedUrlData = await downloadAuthorisation.json();
+      const downloadUrl = `${authData.downloadUrl}/file/${B2_BUCKET_NAME}/${fileName}?Authorization=${signedUrlData.authorizationToken}`;
```

Comment:
This looks a bit untidy - can you split it onto multiple lines more gracefully?

Reply:
Mh, I don't fully agree @CodeKrakken 
I think that is good to have it in one line as it is 1 URL.
To me, the naming is clear enough to make it readable, and I would probably feel a bit more lost if they started to be in separate lines or even more split.
Anyway, just because is clear to me doesn't mean it is for everybody. 
Can you make a example of how it would be written if it were on multiple lines @CodeKrakken ?

I am thinking maybe also something like that can be helpful for you:
```
const b2AuthUrl = authData.downloadUrl;
const authorizationParam =  `Authorization=${signedUrlData.authorizationToken}`;
const downloadUrl = `${b2AuthUrl}/file/${B2_BUCKET_NAME}/${fileName}? authorizationParam`;
```
What do you all think?